backend/apps/core/utils/caching.py

```
@@ -0,0 +1,369 @@
+"""
+Response caching utilities for API endpoints.
+
+Provides caching mixins and utilities to reduce database load and improve
+user experience by serving cached responses even when rate limited.
+"""
+import hashlib
+import json
+from functools import wraps
+from typing import Optional, Any, Set
```

Ruff (0.14.3)

`typing.Set` is deprecated, use `set` instead

(UP035)